packages/react-jss/src/createHoc.js

```diff
@@ -76,8 +60,7 @@ export default function createHOC<
   options: Options
 ): ComponentType<OuterPropsType> {
   const isThemingEnabled = typeof stylesOrCreator === 'function'
```

Comment:
I think we can also get rid of isThemingEnabled, because if a theme is on the context, its weird it would be implicitely depending on styles being a function

Reply:
No, we still need to do some checks based on `isThemingEnabled`

Reply:
Lets revisit them, I think we can get rid of it.

Reply:
We still need this to decide which theme we are using. 

When someone passed a function as the styles, we use the theme prop. 

Because the user still might pass `injectTheme: true` which would subscribe to the theme, even though the styles are not theme dependent, we would create additional stylesheets.

Reply:
I think isThemingEnabled logic needs to be different,  e.g.  `shouldSubscribeTheme = options.injectTheme || stylesIsfunction`  and rendering of styles just based on stylesIsFunction